libs/contrib/Data/SortedMap.idr

```diff
@@ -319,6 +319,44 @@ export
 mergeLeft : SortedMap k v -> SortedMap k v -> SortedMap k v
 mergeLeft = mergeWith const
 
+treeLookupBetween : Ord k => k -> Tree n k v o -> (Maybe (k,v),Maybe (k,v))
+treeLookupBetween k (Leaf k' v) with (k < k')
+  treeLookupBetween k (Leaf k' v) | True = (Nothing, Just (k',v))
+  treeLookupBetween k (Leaf k' v) | False = (Just (k',v), Nothing)
+treeLookupBetween k (Branch2 t1 k' t2) with (k < k')
+  treeLookupBetween k (Branch2 t1 k' t2) | True =
+    let t1r = treeLookupBetween k t1
+    in
+      case t1r of
+        (x,Nothing) => (x,snd $ treeLookupBetween k t2)
```

Comment:
I feel like the logic would be clearer if we had separate `leftMost` and `rightMost`
functions instead of `snd` and `fst` of tree lookups for a key that's clearly too
small or big to be found.

Reply:
Ok, that makes sense. I modified the code and I also exported `leftMost` and `rightMost` since they have easily understood functionality and seem like they might come in handy for users. But if you would rather me to keep them private, let me know.